indexing/generate_index.py

```diff
@@ -0,0 +1,82 @@
+# TODO: Remove "contents"
+#       Add links within the doc (contents in a hierarchy?)
+#       Consider auto-gen on release
+
```

Comment:
I'm getting a few linting errors on this file. Could you do a one-off Black run on it?

Reply:
Done